roles/vdm/tasks/postgres/postgres.yaml

```diff
@@ -5,6 +5,29 @@
     - uninstall
     - update
 
+- name: both internal
+  block:
+    - set_fact:
+        default_postgres_internal: "{{ V4_CFG_POSTGRES_SERVERS.default.internal }}"
```

Comment:
The looping here should be over all postgres_servers. If any one is external then we have to determine how we handle the case of the cds internal database server. A customer may have the retail product in their order but may not deploy that product. Does that keep them from creating a cluster and then deploying other products? The customer will wonder why. Should use the `with_items` block for `for` block to cycle of all V4_CFG_POSTGRES_SERVERS.

Reply:
This code can be removed. There is already a check for this.

Reply:
Ran through couple scenarios with default and CDS postgres, this code block verifies that cds-postgres server is internal if not fails. 
Keeping this block, as it safeguards the cds-postgres internal condition check for both internal and external servers.

Reply:
The code here simply needs to check if any of the `V4_CFG_POSTGERS_SERVERS` is external if the cds database is present. Should be much simpler.

Reply:
Simplified the check and verified the following scenarios with the new code:
**External postgres servers configured:**

| **default postgres** | **cds-postgres** | **Error message**                                             |
| ---------------- | ------------ | --------------------------------------------------------- |
| False            | True         | Yes. Mix and match error                                  |
| True             | False        | Yes. New error msg                                        |
| True             | True         | No. Successful deployment. Internal databases configured. |
| False            | False        | Yes. New error msg                                        |

**Internal database setup. No external postgres server configured:**

| **default postgres** | **cds-postgres** | **Error message**                                                  |
| ---------------- | ------------ | -------------------------------------------------------------- |
| False            | False        | Yes. New error msg                                             |
| True             | False        | Yes. New error msg                                             |
| False            | True         | Yes. Invalid configuration. Default is expected to be internal |
| True             | True         | No. Successful deployment                                      |